engine/security/src/main/scala/pl/touk/nussknacker/ui/security/oidc/OidcAuthenticationConfiguration.scala

```diff
@@ -0,0 +1,77 @@
+package pl.touk.nussknacker.ui.security.oidc
+
+import com.auth0.jwk.{JwkProvider, JwkProviderBuilder}
+import pl.touk.nussknacker.engine.util.config.URIExtensions
+import pl.touk.nussknacker.ui.security.oauth2.{JwtConfiguration, OAuth2Configuration}
+import pl.touk.nussknacker.ui.security.oauth2.ProfileFormat.OIDC
+import sttp.client.{NothingT, SttpBackend}
+import sttp.model.MediaType
+
+import java.net.URI
+import java.security.PublicKey
+import java.util.NoSuchElementException
+import scala.concurrent.{ExecutionContext, Future}
+
+case class OidcAuthenticationConfiguration(usersFile: URI,
+                                           anonymousUserRole: Option[String] = None,
+
+                                           issuer: URI,
+                                           clientId: String,
+                                           clientSecret: Option[String],
+                                           redirectUri: Option[URI] = None,
+                                           accessTokenIsJwt: Boolean = false,
+                                           audience: Option[String] = None,
+
+                                           // The following values are used for overriding the ones obtained
+                                           // from the OIDC Discovery or in case it is not supported at all.
+                                           // They may be relative to the issuer.
+                                           authorizationEndpoint: Option[URI] = None,
+                                           tokenEndpoint: Option[URI] = None,
+                                           userinfoEndpoint: Option[URI] = None,
+                                           jwksUri: Option[URI] = None
+                                          ) extends URIExtensions {
+
+  lazy val oAuth2Configuration: OAuth2Configuration = OAuth2Configuration(
+    usersFile = this.usersFile,
```

Comment:
probably some of this `this.` references are redundant

Reply:
Though, I thought it would be more descriptive in this quite complex transformation. Shall I remove it?